cmd.go

```diff
@@ -498,14 +623,17 @@ func parseTestArguments(args []string) (packages, testFlags []string) {
 		}
 		if strings.HasPrefix(arg, "-") &&
 			!strings.HasPrefix(arg, "-instrument") &&
-			!strings.HasPrefix(arg, "-godebugwork") {
+			!strings.HasPrefix(arg, "-godebugwork") &&
+			!strings.HasPrefix(arg, "-tags") &&
+			!strings.HasPrefix(arg, "-o") &&
+			!strings.HasPrefix(arg, "-c") {
```

Comment:
The `HasPrefix` check breaks if the user tries to pass a flag to their test that shares the same prefix. This was ok for "instrument" and "godebugwork", but there is a pretty good chance of conflict with "o" and "c".

Something like this would be better:

```go
isFlag := func(arg, name string) bool {
    return arg == name || strings.HasPrefix(arg, name + "=")
}
sep := len(args)
for i, arg := range args {
    if strings.HasPrefix(arg, "--") {
        arg = arg[1:]
    }
    if strings.HasPrefix(arg, "-") &&
        !isFlag(arg, "-instrument") &&
        !isFlag(arg, "-godebugwork") &&
        !isFlag(arg, "-tags") &&
        !isFlag(arg, "-o") &&
        !isFlag(arg, "-c") &&
```